classes/Cart.php

```diff
@@ -4011,6 +4011,14 @@ public function getRawSummaryDetails(int $id_lang, bool $refresh = false): array
             'invoice' => AddressFormat::getFormattedLayoutData($invoice),
         ];
 
+        /*
+         * First add all applicable cartRules before calculating the totals and the discounts
+         */
+        if (CartRule::isFeatureActive() && $this->id) {
```

Comment:
I'm not sure it's the best way 🤔  But I understand your issue that the cart rules have not been updated. Maybe you have a simpler solution here if you call:

```php
$cartRules = $this->getCartRules();
```

right here instead of L4051, then they also should be automatically added.

Then on L4051 you can do:

```php
'discounts' => array_values($cartRules),
```

Reply:
But as you mentioned in the issue, this is just a big ugly patch, the problem is not that the `getRawSummaryDetails` doesn't add the cart rules when expected (it'snot its job). The problem comes from the fact that the created cart rule in the create order page was never added to the cart in the first place.

You should fix the problem at the source (adding a cart rule correctly to a cart), not at the output.

Reply:
There are two ways to fix this correctly:
- the URL in the iframe should not be a generic independent "Cart rule form", but a dedicated controller "Add cart rule to a cart", that handles to add the cart rule to a cart right after it's been created
- keep using the generic cart rule form, but then you ned to get back the generated cart rule ID, and add it to the cart via `this.cartRuleManager.addCartRuleToCart(cartRuleId, <number> this.cartId);`